fiftyone/operators/panel.py

```diff
@@ -118,39 +141,38 @@ class WriteOnlyError(Exception):
     """Error raised when trying to read a write-only property."""
 
 
-class PanelRefBase:
-    """
-    Base class for panel state and data.
+class PanelRefBase(object):
+    """Base class for panel state and data.
 
-    Attributes:
-        _data (dict): A dictionary to store the data or state.
-        _ctx: The context object containing the operations.
+    Args:
+        ctx: an :class:`fiftyone.operators.executor.ExecutionContext`
     """
 
     def __init__(self, ctx):
         self._data = {}
         self._ctx = ctx
 
-    def set(self, key, value):
-        """
-        Sets the value in the dictionary.
+    def set(self, key, value=None):
+        """Sets some value(s) in the dictionary.
 
         Args:
-            key (str): The key.
-            value (any): The value.
+            key: a key, ``"nested.key.path"``, or dict mapping multiple
+                possibly-nested keys to values
+            value (None): the value, if key is a string
         """
-        pydash.set_(self._data, key, value)
+        d = {key: value} if value is not None else key
```

Comment:
This should be something like:

```python
d = key if isinstance(key, dict) else {key: value}
```

Since `set('foo', None)` is valid.

Reply:
ah good catch!